### PR TITLE
chore: added restart actions for dbs/apps/drains

### PR DESCRIPTION
**Pull request by madhuravius**
~TODO - disable on v4 log drains as well (restart not allowed)~

---

* ~proof~
* ~tests~ (adding to part 2 with deletes)

image

image

image

delete works w/ confirm

image

image

image

if v4, would look like this:

image

### CODE REVIEW

**Comment by joeclayallday** [comment]
Pushed up a small design tweak @madhuravius 
Screen Shot 2023-08-23 at 8 39 06 PM

File: src/ui/pages/app-detail-settings.tsx [hunk 1]
```diff
@@ -76,6 +79,16 @@ export const AppSettingsPage = () => {
     navigate(appActivityUrl(id));
   };
 
+  const restartAction = restartApp({ id });
+  const restartLoader = useLoader(restartAction);
+  const submitRestart = (e: SyntheticEvent) => {
```
Comment (neurosnap) [inline src/ui/pages/app-detail-settings.tsx:84]: Use the actual event type here.

File: src/ui/pages/app-detail-settings.tsx [hunk 2]
```diff
@@ -157,15 +170,32 @@ export const AppSettingsPage = () => {
...
+              disabled={
+                isUpdating ||
+                updatingAppLoader.isLoading ||
+                restartLoader.isLoading
+              }
             >
-              {isUpdating || updatingAppLoader.isLoading
+              {isUpdating ||
+              updatingAppLoader.isLoading ||
+              restartLoader.isLoading
```
Comment (neurosnap) [inline src/ui/pages/app-detail-settings.tsx:198]: `Button` has `isLoading` which will: disable the button and set the "Loading ..." text.

File: src/ui/pages/app-detail-settings.tsx [hunk 3]
```diff
@@ -157,15 +170,32 @@ export const AppSettingsPage = () => {
...
+            {isDeprovisioning ? "Restarting..." : "Restart"}
+          </Button>
+
           <hr className="mt-6" />
 
           <div className="flex mt-4">
             <Button
               className="w-40 flex semibold"
               type="submit"
-              disabled={isUpdating || updatingAppLoader.isLoading}
```
Comment (neurosnap) [inline src/ui/pages/app-detail-settings.tsx:166]: What is the purpose of `isUpdating`?  The loaders are what indicate if something is being updated so this seems unnecessary.

File: src/ui/pages/app-detail-settings.tsx (hunk 2 above)
Comment (madhuravius) [reply src/ui/pages/app-detail-settings.tsx:198]: got it

File: src/ui/pages/app-detail-settings.tsx (hunk 3 above)
Comment (madhuravius) [reply src/ui/pages/app-detail-settings.tsx:166]: this is an older component despite you and me working on update before loaders were as popular as they are now in this codebase (when you were OOO i think and i made stuff up as i went)

**Review by joeclayallday — APPROVED** [review]

**Review by almathew — APPROVED** [review]

**Pull request merged**